connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/resources/ConnectorsResource.java

```diff
@@ -99,7 +100,7 @@ public Response createConnector(final @QueryParam("forward") Boolean forward,
         herder.putConnectorConfig(name, configs, false, cb);
         Herder.Created<ConnectorInfo> info = completeOrForwardRequest(cb, "/connectors", "POST", createRequest,
                 new TypeReference<ConnectorInfo>() { }, new CreatedConnectorInfoTranslator(), forward);
-        return Response.created(URI.create("/connectors/" + name)).entity(info.result()).build();
+        return Response.created(URI.create("/connectors/" + URLEncoder.encode(name, "UTF-8"))).entity(info.result()).build();
```

Comment:
This line is getting a little busy. You might consider pulling the encoding logic into a separate method (especially because addressing @soenkeliebau's concerns might result in more code to do this) and then adding a local variable for the encoded name. This same method could be used on line 149, too.